openssl-dynamic/src/main/c/io_netty_tcnative_jni_ocsp_Native.c

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <jni.h>
+#include <openssl/ssl.h>
+#include <openssl/tls1.h>
+#include <stdio.h>
+
+#include <tcn.h>
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-variable"
+#include <ssl_private.h>
+#pragma GCC diagnostic pop
+
+#include <io_netty_tcnative_jni_ocsp_Native_init.h>
+#include <io_netty_tcnative_jni_ocsp_Native.h>
+
+/**
+ * https://www.openssl.org/docs/man1.0.2/ssl/SSL_set_tlsext_status_type.html
+ */
+
+/**
+ * Some globally cached values.
+ */
+static jclass CALLBACK_CLAZZ = NULL;
+static jmethodID CALLBACK_METHOD = NULL;
+
+static void logException(JNIEnv *env) {
+  (*env)->ExceptionDescribe(env);
+  (*env)->ExceptionClear(env);
+}
+
+/**
+ * Frees the callback
+ */
+static void freeCallback(JNIEnv *env, tcn_ssl_ctxt_t *c, jweak callbackRef) {
+  
+  if (c != NULL) {
+    SSL_CTX_set_tlsext_status_arg(c->ctx, NULL);
+    SSL_CTX_set_tlsext_status_cb(c->ctx, NULL);
+  }
+
+  (*env)->DeleteWeakGlobalRef(env, callbackRef);
+}
+
+/**
+ * See:
+ *   io_netty_tcnative_jni_ocsp_client_callback
+ *   io_netty_tcnative_jni_ocsp_server_callback
+ */
+static int io_netty_tcnative_jni_ocsp_callback(int errorCode, SSL *ssl, void *arg) {
+
+  // This is conceptually not possible.
+  if (CALLBACK_CLAZZ == NULL) {
```

Comment:
is this check necessary? the `CALLBACK_METHOD` check should be sufficient right and `CALLBACK_CLAZZ` is not directly used in this method.